packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -23,6 +24,63 @@ export interface IService extends IResource {
   readonly serviceArn: string;
 }
 
+/**
+ * Properties for ECS container targets.
+ */
+export interface TargetProps {
+  /**
+   * Name of the container.
+   */
+  readonly containerName: string,
+
+  /**
+   * Container targets of the container.
+   */
+  readonly containerTargets: ContainerTargetProps[]
```

Comment:
Not sure I like this name. It feels like introducing a new concept to users that I wonder whether it's worthwhile.

Also: ultimately, we just need a flat list of `(container, port, targetGroupName, listener)`, so why introduce all this hierarchy here? Feels like a flat list of tuples would give a much more convenient API.

Reply:
Yeah I agree, since for each service for most of cases there will only be one or two targets to be registered. Using a flat list should be a better option.